twopaneview/src/TwoPaneView.tsx

```diff
@@ -3,57 +3,54 @@ import {
   View,
   Dimensions,
 } from 'react-native';
-import { DualScreenInfo } from 'react-native-dualscreeninfo'
+import { DualScreenInfo, DualScreenInfoPayload, WindowRect, DeviceRotation } from 'react-native-dualscreeninfo'
 import {Orientation, PanePriority, PaneMode} from "./types"
 
-type Props = {
-} & Partial<DefaultProps>;
-
-type DefaultProps = Readonly<typeof defaultProps>;
-
-const defaultProps = { panePriority: PanePriority.Pane1, 
-  panePriorityVerticalSpanning: Orientation.Horizontal, 
-  paneMode: PaneMode.Auto
-}
-
 type State = {
-  dims: any,
+  windowRects: WindowRect[],
   spanning: boolean, 
-  panePriority?: string,
-  panePriorityVerticalSpanning?: string,
-  paneMode?: string,
-};
+  rotation: DeviceRotation,
+}
+
+export interface TwoPaneViewProps {
+  panePriority: PanePriority,
+  paneMode: PaneMode,
+}
 
-export class TwoPaneView extends Component<Props, State> {
+export class TwoPaneView extends Component<TwoPaneViewProps, State> {
   state: State = {
-    dims: Dimensions.get('window'),
+    windowRects: DualScreenInfo.windowRects,
```

Comment:
The current implementation does not seem to use windowRects anymore.
Might want to consider taking them out.